source/jobs/JobEngine.cpp

```diff
@@ -270,7 +270,7 @@ int JobEngine::exec_cmd(string operation, PlainJobDocument::JobAction action)
 
     int execResult;
     int returnCode;
-    int pid = vfork();
+    int pid = fork();
```

Comment:
Child process created by`vfork()` suspends execution of parent process until child process completes its execution  wereas, both parent and the child process created using`fork()` start their execution from next statement after fork() and both processes get executed simultaneously.

This might result in executing multiple job executions simultaneously. 

Device Client anyways does not work on Mac OS so I would not encourage making this change until we make the software MacOS compatible.

Reply:
The child and parent process do different things. The parent process waits for the child process to finish already.

Reply:
I might have had a wrong understanding then. @timmattison let me test these changes locally and review this change again.